ttnn/cpp/ttnn/operations/reduction/generic/device/reduce_op.cpp

```diff
@@ -77,25 +79,34 @@ std::vector<tt::tt_metal::LegacyShape> Reduce::compute_output_shapes(const std::
             padding[3] = Padding::PadDimension{0, 31};
             break;
     }
-    return {tt::tt_metal::LegacyShape(output_shape, padding)};
+    auto output2 = tt::tt_metal::LegacyShape(output_shape, padding);
+    std::cout << output2 << std::endl;
+    std::cout << output2.logical_shape() << std::endl;
+    return {tt::tt_metal::LegacyShape(output_shape, padding).logical_shape()};
 }
 
-std::vector<Tensor> Reduce::create_output_tensors(const std::vector<Tensor>& input_tensors) const {
-    const auto& input_tensor = input_tensors.at(0);
-    if (this->output_mem_config.is_sharded()) {
-        auto output_shape = this->compute_output_shapes(input_tensors).at(0);
-        auto shard_spec = input_tensor.shard_spec().value();
-        shard_spec.shape[0] = tt_metal::compute_volume(output_shape) / output_shape[-1];
-        auto mem_config = this->output_mem_config;
-        mem_config.shard_spec = shard_spec;
-        return {
-            create_device_tensor(output_shape, this->output_dtype, Layout::TILE, input_tensor.device(), mem_config)};
-    } else {
-        return operation::generic_create_output_tensors(
-            *this, input_tensors, this->output_dtype, Layout::TILE, this->output_mem_config);
-    }
+std::vector<TensorLayout> Reduce::compute_output_layouts(const std::vector<Tensor>& input_tensors) const {
+    return {TensorLayout::fromLegacyPaddedShape(this->output_dtype, PageConfig(Layout::TILE), this->output_mem_config, input_tensors.at(0).get_padded_shape())};
```

Comment:
This needs to be updated to use output padded shape. Maybe something like:
- Take alignment from non-tile dims from input
- Compute alignment with tile + output shapes